File: backend-spring/src/main/java/com/cronos/entity/Payment.java
package com.cronos.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.SQLDelete;
import org.hibernate.annotations.SQLRestriction;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
@Table(name = "payments")
@SQLDelete(sql = "UPDATE payments SET deleted_at = NOW() WHERE id = ?")
@SQLRestriction("deleted_at IS NULL")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class Payment {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(columnDefinition = "uuid")
    private UUID id;

    @JsonProperty("order_id")
    @Column(name = "order_id", nullable = false, unique = true, columnDefinition = "uuid")
    private UUID orderId;

    @Column(nullable = false, precision = 15, scale = 2)
    private BigDecimal amount;

    @Column(length = 50)
    private String method;

    @Column(length = 50)
    @Builder.Default
    private String status = "pending";

    @JsonProperty("paid_at")
    @Column(name = "paid_at")
    private LocalDateTime paidAt;

    @JsonProperty("created_at")
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @JsonProperty("updated_at")
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @JsonIgnore
    @Column(name = "deleted_at")
    private LocalDateTime deletedAt;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "order_id", insertable = false, updatable = false)
    @JsonIgnore
    private CronosOrder order;

    @OneToOne(mappedBy = "payment", fetch = FetchType.LAZY)
    @JsonProperty("midtrans_transaction")
    private MidtransTransaction midtransTransaction;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: fluttermultiplatform/lib/models/payment.dart
class Payment {
  final String id;
  final String orderId;
  final double amount;
  final String method;
  final String status;
  final String? paidAt;
  final String? snapUrl;

  Payment({required this.id, this.orderId = '', this.amount = 0, this.method = '', this.status = '', this.paidAt, this.snapUrl});

  factory Payment.fromJson(Map<String, dynamic> json) => Payment(
        id: json['id']?.toString() ?? '',
        orderId: json['order_id']?.toString() ?? '',
        amount: (json['amount'] ?? 0).toDouble(),
        method: json['method'] ?? '',
        status: json['status'] ?? '',
        paidAt: json['paid_at'],
        snapUrl: json['midtrans_transaction'] != null ? json['midtrans_transaction']['snap_url'] : json['snap_url'],
      );
}
